### PR TITLE
LB-71: Token regeneration

### PR DESCRIPTION
**Pull request by paramsingh**
Work continued from #62. 

Did a rebase from current master. I think I have fixed most of the stuff talked about in the comments there.

Also added a test for the update_token function.

### CODE REVIEW

File: webserver/templates/user/resettoken.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{% extends 'base.html' %}
+
+{% block title %}Reset token - ListenBrainz{% endblock %}
+
+{% block content %}
+  <h3 class="page-title">Reset token</h3>
+  <p>Are you sure you want to reset your token? </p>
+
+  <p>If you do, you will need to replace your old token with the newly generated token in any of your applications that use your token. The old token will stop working immediately. Be sure to update the token immediately in your applications to ensure that you have no disruptions in recording your listens.</p>
```
Comment (mayhem) [inline webserver/templates/user/resettoken.html]: Well said!

File: db/user.py [hunk 2]
```diff
@@ -23,6 +23,22 @@ def create(musicbrainz_id):
         })
         return result.fetchone()["id"]
 
+def update_token(id):
+    """Update a user's token to a new UUID
+
+    Args:
+        id (int) - the row id of the user to update
+    """
+    with db.engine.connect() as connection:
```
Comment (mayhem) [inline db/user.py]: You're not catching any possible DB exceptions here. As per usual, catch exceptions and then show an appropriate error message and log an error as well.

**Comment by paramsingh** [comment]
Added error catching in the db code.

File: webserver/templates/user/resettoken.html (hunk 1 above)
Comment (paramsingh) [inline webserver/templates/user/resettoken.html]: Haha, I can't take credit for this.

https://github.com/metabrainz/listenbrainz-server/pull/62#r42362760

**Pull request merged**